src/Vehicle/Vehicle.cc

```diff
@@ -3421,17 +3421,25 @@ void Vehicle::_handleMavlinkLoggingData(mavlink_message_t& message)
 {
     mavlink_logging_data_t log;
     mavlink_msg_logging_data_decode(&message, &log);
-    emit mavlinkLogData(this, log.target_system, log.target_component, log.sequence,
-                        log.first_message_offset, QByteArray((const char*)log.data, log.length), false);
+    if (static_cast<size_t>(log.length) > sizeof(log.data)) {
+            qWarning() << "Invalid length for LOGGING_DATA, discarding." << log.length;
```

Comment:
Sorry, I don't like to be nitpicky about this, but this is still off (below as well).